packages/core-ethereum/src/ethereum.ts

```diff
@@ -42,7 +43,7 @@ export async function createChainWrapper(
   const wallet = new Wallet(privateKey).connect(provider)
   const publicKey = PublicKey.fromPrivKey(privateKey)
   const address = publicKey.toAddress()
-  const providerChainId = await provider.getNetwork().then((res) => res.chainId)
+  const providerChainId = (await provider.getNetwork()).chainId
```

Comment:
```suggestion
  const providerChainId = (await provider.getNetwork())?.chainId
```

Reply:
I actually want this to fail if the environment does not exist or if the network does not exist